### PR TITLE
Fix hashjoin detection and symbol extraction

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: sql/src/test/java/io/crate/planner/operators/HashJoinDetectorTest.java [hunk 1]
```diff
@@ -82,4 +82,16 @@ public void testNotPossibleIfNotAnInnerJoin() {
...
+    @Test
+    public void testNotPossibleOnEqWithoutRelationFieldsOnBothSides() {
+        Symbol joinCondition = SQL_EXPRESSIONS.asSymbol("t1.x = 4");
+        assertThat(HashJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
+    }
+
+    @Test
+    public void testNotPossibleOnNotWrappingEq() {
+        Symbol joinCondition = SQL_EXPRESSIONS.asSymbol("NOT (t1.a = t2.b)");
+        assertThat(HashJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
```
Comment (seut) [inline sql/src/test/java/io/crate/planner/operators/HashJoinDetectorTest.java]: I think `t1.a = t2.b AND NOT(t1.x = t2.y)` can be done using the hash join. please add a test

Comment (matriv) [reply sql/src/test/java/io/crate/planner/operators/HashJoinDetectorTest.java]: @seut there is a test here: https://github.com/crate/crate/pull/7227/files/d3577eedb6e2028abf61c1fe8c3a4384cb6002b3#diff-8316d16a3c5b81301269e8e56656c6b5R46

Should we add a specific one like you suggested?

Comment (seut) [reply sql/src/test/java/io/crate/planner/operators/HashJoinDetectorTest.java]: this the existing test is also fine, didn't see it, sry

Comment (matriv) [reply sql/src/test/java/io/crate/planner/operators/HashJoinDetectorTest.java]: ok, no problem!

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Pull request merged**